src/c/compat/backports_zstd_edits.h

```diff
@@ -45,4 +45,54 @@ BACKPORTSZSTD__PyArg_UnpackKeywords(
 #define BACKPORTSZSTD__PyNumber_Index(o) \
     _PyNumber_Index(o)
 
+#if PY_VERSION_HEX < 0x030D0000 // Python 3.12 and below
+#define BACKPORTSZSTD_PyMutex PyThread_type_lock
```

Comment:
Since `PyThread_type_lock` is not always backed by a PyMutex (though that may change in 3.15), perhaps it would be better to call it a generic name like `BACKPORTS_ZSTD_LOCK`?